autoscale-scheduler/cmd/main.go

```diff
@@ -53,12 +53,14 @@ func runProgram(logger *zap.Logger) (err error) {
 	// services and be able to more coherently wait for shutdown
 	// without needing a sleep.
 	ctx, cancel := signal.NotifyContext(context.Background(), syscall.SIGTERM)
-	defer cancel()
 	ctx = srv.SetShutdownSignal(ctx)
 	ctx = srv.WithOrchestrator(ctx)
 	ctx = srv.SetBaseContext(ctx)
 	orca := srv.GetOrchestrator(ctx)
-	defer func() { err = orca.Service().Wait() }()
+	defer func() {
+		cancel()
+		err = orca.Service().Wait()
```

Comment:
I am struggling to understand how is it supposed to work.

Looking into the implementation of `orca.Service().Wait()` I see it never uses the context passed via `srv.GetOrchestrator(ctx)`, do I read it wrong? In Wait (and in other places of the orca lib) it uses internal.BackgroundContext which is defined as context.Background and from the top of my head I don't see internal.BackgroundContext be overriden?

Reply:
I don't see nothing like that in the code, I think. 

It does the opposite: creates a context with orchestrator in the context available by key `orchestratorCtxKey{}` (basically, orca variable is the object).